packages/vite/src/node/optimizer/optimizer.ts

```diff
@@ -496,5 +505,85 @@ export async function initDepsOptimizer(
     }, timeout)
   }
 
+  const runOptimizerIfIdleAfterMs = 100
+
+  let registeredIds: { id: string; done: () => Promise<any> }[] = []
+  let seenIds = new Set<string>()
+  let workersSources = new Set<string>()
+  let waitingOn: string | undefined
+
+  function resetRegisteredIds() {
+    registeredIds = []
+    seenIds = new Set<string>()
+    workersSources = new Set<string>()
+    waitingOn = undefined
+  }
+
+  function registerWorkersSource(id: string): void {
+    workersSources.add(id)
+    if (waitingOn === id) {
+      waitingOn = undefined
+    }
+  }
+
+  function delayDepsOptimizerUntil(id: string, done: () => Promise<any>): void {
+    if (!depsOptimizer.isOptimizedDepFile(id) && !seenIds.has(id)) {
+      seenIds.add(id)
+      registeredIds.push({ id, done })
+      runOptimizerWhenIdle()
+    }
+    if (server && !optimizeDepsEntriesVisited) {
+      optimizeDepsEntriesVisited = true
+      pretransformOptimizeDepsEntries(server)
+    }
+  }
+
+  function runOptimizerWhenIdle() {
+    if (!waitingOn) {
+      const next = registeredIds.pop()
+      if (next) {
+        waitingOn = next.id
+        const afterLoad = () => {
+          waitingOn = undefined
+          if (registeredIds.length > 0) {
+            runOptimizerWhenIdle()
+          } else if (!workersSources.has(next.id)) {
+            getDepsOptimizer(config)?.run()
+          }
+        }
+        next
+          .done()
+          .then(() => {
+            setTimeout(
+              afterLoad,
+              registeredIds.length > 0 ? 0 : runOptimizerIfIdleAfterMs
+            )
+          })
+          .catch(afterLoad)
+      }
+    }
+  }
+
   return depsOptimizer
 }
+
+export async function pretransformOptimizeDepsEntries(
```

Comment:
```suggestion
export async function preTransformOptimizeDepsEntries(
```

I think this should be called `preTransform` as the config is`config.server.preTransformRequests` too